pyccl/ccl_sigM.i

```diff
@@ -0,0 +1,59 @@
+%module ccl_massfunc
+
+%{
+/* put additional #include here */
+%}
+
+%include "../include/ccl_massfunc.h"
+
+// Enable vectorised arguments for arrays
+%apply (double* IN_ARRAY1, int DIM1) {(double* logM, int nM)};
+%apply (int DIM1, double* ARGOUT_ARRAY1) {(int nout, double* output)};
+
+/* The python code here will be executed before all of the functions that
+   follow this directive. */
+%feature("pythonprepend") %{
+    if numpy.shape(logM) != (nout,):
+        raise CCLError("Input shape for `halo_mass` must match `(nout,)`!")
+%}
+
+%inline %{
+
+void sigM_vec(ccl_cosmology * cosmo, double a,
+	      double *logM, int nM,
+	      int nout, double* output, int *status)
+{
+  // Check if sigma has already been calculated
+  if (!cosmo->computed_sigma) {
+    *status = CCL_ERROR_SIGMA_INIT;
+    ccl_cosmology_set_status_message(
+      cosmo,
+      "ccl_massfunc.c: ccl_sigmaM(): linear power spctrum has not been computed!");
+    return;
+  }
+
+  for(int i=0; i<nM; i++)
+    output[i] = ccl_sigmaM(cosmo, logM[i], a, status);
+}
+
+void dlnsigM_dlogM_vec(ccl_cosmology * cosmo,
+		       double *logM, int nM,
+		       int nout, double* output, int *status)
+{
+  // Check if sigma has already been calculated
+  if (!cosmo->computed_sigma) {
+    *status = CCL_ERROR_SIGMA_INIT;
+    ccl_cosmology_set_status_message(
+      cosmo,
+      "ccl_massfunc.c: ccl_sigmaM(): linear power spctrum has not been computed!");
+    return;
+  }
```

Comment:
Ditto here for code I think you meant to delete.

Reply:
(somehow I lost your previous comment, not sure what happened there)
This function is useful here to translate the output of `ccl_sigmaM` into a numpy array.
I've removed the check for the linear power spectrum, since that one is superfluous.

Reply:
I just meant the check, not the whole function.

Reply:
Ah, ok. Yes, you're right. I've removed both instances.